src/App.tsx

```diff
@@ -92,20 +102,20 @@ const ColorField: Component<{
     <div class="flex gap-1">
       <input
         type="text"
-        class="text-white w-20 rounded shadow-inner shadow-black bg-transparent focus:bg-zinc-950 border border-zinc-600 focus:border-sky-500 outline-0 px-1 font-mono text-sm"
+        class="w-20 rounded border border-zinc-600 bg-transparent px-1 font-mono text-sm text-white shadow-inner shadow-black outline-0 focus:border-sky-500 focus:bg-zinc-950"
         value={props.value}
         onchange={(e) => props.update(e.currentTarget.value)}
-      // TODO: validate before calling update to ensure it's valid hex
+        // TODO: validate before calling update to ensure it's valid hex
       />
       <input
         id={props.property}
-        class="text-white w-8 rounded shadow-inner shadow-black bg-transparent focus:bg-zinc-950 border border-zinc-600 focus:border-sky-500 outline-0 px-1"
+        class="w-8 rounded border border-zinc-600 bg-transparent px-1 text-white shadow-inner shadow-black outline-0 focus:border-sky-500 focus:bg-zinc-950"
         type="color"
         value={props.value}
         onchange={(e) => {
```

Comment:
trying to figure out why this breaks if I switch it to `oninput`. my assumption was that this would work now that updates are being diffed. instead, it switches to a text field for some reason, even if the output value is valid hex.

the text input event handler probably does need to remain `onchange` since intermediate values are unlikely to be perfect hex.

Reply:
Ooh this was an interesting bug, good find! Pushed a fix for this to main https://github.com/sebastiansandqvist/json-gui/commit/7eb35e672573afbb5cf7bee31bb189b05268aa82. Turns out you can't reuse some regex like `const hexRegex = /^#[a-f0-9]{6}$/gi;` because it has state. If the previous `.test` succeeded, the next will fail.